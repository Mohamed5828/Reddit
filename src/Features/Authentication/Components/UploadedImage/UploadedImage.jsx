// Import styled components
import {
  DeleteButton,
  StyledUploadedImage,
  Thumb,
  ThumbInner,
} from "./UploadedImage.styled";

// Import icons
import { ImCross } from "react-icons/im";

/**
 * Uploaded image component in images & videos tab
 *
 * @param {File} file - The uploaded image
 * @param {Function} deleteFile - The function to delete the image
 * @param {Boolean} isSelected - Boolean to check if the image is selected
 * @param {Number} selectedImageId - The id of the selected image
 * @param {Number} id - The id of this image
 * @param {Function} handleClick - The function to handle the click event on image
 * @returns {React.Component} - Uploaded image component in images & videos tab
 */
const UploadedImage = ({ file, deleteFile, handleClick, open }) => {
  return (
    <Thumb
      onClick={() => {
        open();
      }}
    >
      <ThumbInner>
        <StyledUploadedImage
          thumbnail={true}
          selected={true}
          src={file.preview}
          // Revoke data uri after image is loaded
          onLoad={() => {
            URL.revokeObjectURL(file.preview);
          }}
          alt="uploaded preview"
        />
        <DeleteButton
          variant="danger"
          onClick={(e) => {
            e.stopPropagation();
            deleteFile(file);
          }}
        >
          <ImCross size={20} />
        </DeleteButton>
      </ThumbInner>
    </Thumb>
  );
};

export default UploadedImage;
