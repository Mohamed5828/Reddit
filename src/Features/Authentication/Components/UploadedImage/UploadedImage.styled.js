// Import styled
import styled from "styled-components";

// Import bootstrap components
import { Image } from "react-bootstrap";

export const StyledUploadedImage = styled(Image)`
  width: 100%;
  min-width: 100%;
  margin: auto;
  object-fit: cover;
  display: inline-block;
  border-radius: 4px;
  border: ${({ selected }) =>
    selected ? `2px solid ${({ theme }) => theme.color.muted}` : "none"};
  opacity: ${({ selected }) => (selected ? "1" : "0.5")};
  background-color: ${({ theme }) => theme.background.primary};
  &:hover {
    opacity: 1;
  }
`;

export const Thumb = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  &:hover {
    cursor: grab;
  }
`;

export const ThumbInner = styled.div`
  display: flex;
  min-width: 0;
  overflow: hidden;
  margin: 0 12px 12px 0;
  position: relative;
  &:hover {
    button {
      display: inline-flex;
    }
  }
`;

export const DeleteButton = styled.button`
  height: 22px;
  width: 22px;
  border-radius: 50%;
  display: none;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.color.primary};
  color: ${({ theme }) => theme.background.primary};
  position: absolute;
  right: 8px;
  top: 8px;
  padding: 4px;
`;
