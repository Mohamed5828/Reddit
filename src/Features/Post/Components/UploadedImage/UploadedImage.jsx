// Import styled components
import {
  DeleteButton,
  StyledUploadedImage,
  Thumb,
  ThumbInner,
} from "./UploadedImage.styled";

// Import icons
import { ImCross } from "react-icons/im";

// Import hooks
import React, { useEffect } from "react";

/**
 * Uploaded image component in images & videos tab
 *
 * @param {File} file - The uploaded image
 * @param {Function} deleteFile - The function to delete the image
 * @param {Boolean} isSelected - Boolean to check if the image is selected
 * @param {Number} selectedImageId - The id of the selected image
 * @param {Number} id - The id of this image
 * @param {Function} handleClick - The function to handle the click event on image
 * @returns {React.Component} - Uploaded image component in images & videos tab
 */
const UploadedImage = ({
  file,
  deleteFile,
  isSelected,
  selectedImageId,
  id,
  handleClick,
}) => {
  useEffect(() => {
    // When image first render mark it as selected
    handleClick(id);
  }, []);
  return (
    <Thumb onClick={() => handleClick(id)}>
      <ThumbInner>
        <StyledUploadedImage
          className="uploaded-image"
          thumbnail={isSelected && selectedImageId === id}
          selected={isSelected && selectedImageId === id}
          as={file.type.toLowerCase().includes("video") ? "video" : "img"}
          src={file.src}
          // Revoke data uri after image is loaded
          onLoad={() => {
            URL.revokeObjectURL(file.preview);
          }}
          alt="uploaded preview"
        />
        <DeleteButton
          className="delete-img-danger"
          variant="danger"
          onClick={() => deleteFile(file)}
        >
          <ImCross size={20} />
        </DeleteButton>
      </ThumbInner>
    </Thumb>
  );
};

export default UploadedImage;
