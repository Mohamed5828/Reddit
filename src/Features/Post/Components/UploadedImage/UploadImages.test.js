import { shallow } from "enzyme";
import React from "react";

// Import components
import UploadedImage from "./UploadedImage";

describe("Uploaded images container", () => {
  let file = { name: "image", uploadDate: "2020-10-10",type:"image/png" };
  const deleteFile = () => console.log("delete");
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<UploadedImage file={file} deleteFile={deleteFile} />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
