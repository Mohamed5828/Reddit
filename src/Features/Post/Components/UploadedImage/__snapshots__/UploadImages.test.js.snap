// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploaded images container should render without crashing 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div>
    <Styled(Image)
      alt="uploaded preview"
      as="img"
      onLoad={[Function]}
    />
    <styled.button
      onClick={[Function]}
      variant="danger"
    >
      <ImCross
        size={20}
      />
    </styled.button>
  </styled.div>
</styled.div>
`;
